Source/SlideshowView.swift

```diff
@@ -0,0 +1,95 @@
+import UIKit
+
+class SlideshowView: UIView, ViewableControllerContainer {
+    weak var dataSource: ViewableControllerContainerDataSource?
+    weak var delegate: ViewableControllerContainerDelegate?
+    fileprivate unowned var parentController: UIViewController
+    fileprivate var currentPage: Int
+    fileprivate var currentController: ViewableController?
+
+    fileprivate lazy var timer: Timer = {
+        let timer = Timer(timeInterval: 6, target: self, selector: #selector(loadNext), userInfo: nil, repeats: true)
+
+        return timer
+    }()
+
+    init(frame: CGRect, parentController: UIViewController, initialPage: Int) {
+        self.parentController = parentController
+        self.currentPage = initialPage
+
+        super.init(frame: frame)
+
+        self.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+    }
+
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func configure() {
+        self.subviews.forEach { view in
+            view.removeFromSuperview()
+        }
+
+        self.loadPage(self.currentPage, animated: false, isInitial: true)
+    }
+
+    fileprivate func loadPage(_ page: Int, animated: Bool, isInitial: Bool) {
+        let numPages = self.dataSource?.numberOfPagesInViewableControllerContainer(self) ?? 0
+        if page >= numPages || page < 0 {
+            return
+        }
+
+        guard let controller = self.dataSource?.viewableControllerContainer(self, controllerAtIndex: page) as? ViewableController, controller.view.superview == nil else { return }
```

Comment:
So it get's the controller at the indexPath of the collectionview form the superview? Or am I mistaken?